pkgs/applications/networking/cluster/terraform-provider-libvirt/default.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv, buildGoPackage, fetchFromGitHub, libvirt, pkgconfig, makeWrapper, cdrtools }:
+
+# USAGE:
+# install the following package globally or in nix-shell:
+#
+#   (terraform.withPlugins (old: [terraform-provider-libvirt]))
+#
+# configuration.nix:
+#
+#   virtualisation.libvirtd.enable = true;
+#
+# terraform-provider-libvirt does not manage pools at the moment:
+#
+#   $ virsh --connect "qemu:///system" pool-define-as default dir - - - - /var/lib/libvirt/images
+#   $ virsh --connect "qemu:///system" pool-start default
+#
+# pick an example from (i.e ubuntu):
+# https://github.com/dmacvicar/terraform-provider-libvirt/tree/master/examples
+
+buildGoPackage rec {
+  name = "terraform-provider-libvirt-${version}";
+  version = "0.3";
+
+  goPackagePath = "github.com/dmacvicar/terraform-provider-libvirt";
+
+  src = fetchFromGitHub {
+    owner = "dmacvicar";
+    repo = "terraform-provider-libvirt";
+    rev = "v${version}";
+    sha256 = "004gxy55p5cf39f2zpah0i2zhvs4x6ixnxy8z9v7314604ggpkna";
+  };
+
+  buildInputs = [ libvirt pkgconfig makeWrapper ];
+
+  goDeps = ./deps.nix;
+
+  # FIXME It needs mkisofs from cdrtools.
+  # However it needs to be in the PATH of terraform and not just terraform-provider-libvirt
+  # Ideally terraform's plugin wrapper would pick downstream dependencies up
+  propagatedUserEnvPkgs = [ cdrtools ];
+  propagatedBuildInputs = [ cdrtools ];
```

Comment:
cc @zimbatm it might in future to add a mechanism `terraform.withPlugins` to provide this functionality.